### PR TITLE
feat: support auto upload release manifests

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolves: https://github.com/envoyproxy/gateway/issues/311
Signed-off-by: Xunzhuo <bitliu@tencent.com>

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -45,4 +45,14 @@ jobs:
           EG_TAG_VERSION: ${{ steps.tag_env.outputs.version}}
         run: make image.push.multiarch TAG=${EG_TAG_VERSION} IMAGE_NAME=gateway PLATFORMS="linux_amd64 linux_arm64"
 
+      - name: Generate Release Manifests
+        run: make release-manifests
+
+      - name: Upload Release Manifests
+        uses: svenstaro/upload-release-action@v2
```
Comment (arkodg) [inline .github/workflows/release.yaml:52]: is this an alternative to create a release and uploading a release asset using the standard actions ?
https://github.com/actions/create-release#example-workflow---create-a-release

Comment (Xunzhuo) [reply .github/workflows/release.yaml:52]: I will take a look and figure it out, I used this action before and it worked well:)

Comment (lizan) [reply .github/workflows/release.yaml:52]: @arkodg the official `create-release` action is unmaintained and not recommended: https://github.com/actions/create-release#github-action---releases-api

Comment (Xunzhuo) [reply .github/workflows/release.yaml:52]: I am going to try this https://github.com/softprops/action-gh-release in recommendation list.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -45,4 +45,14 @@ jobs:
...
 
+      - name: Generate Release Manifests
+        run: make release-manifests
+
+      - name: Upload Release Manifests
+        uses: softprops/action-gh-release@v1
+        with:
+          files: |
+            release/gatewayapi-crds.yaml
+            release/install.yaml
```
Comment (danehans) [inline .github/workflows/release.yaml:56]: Update `release` to `release-artifacts`, [xref](https://github.com/envoyproxy/gateway/pull/300#discussion_r959064129).

Comment (arkodg) [reply .github/workflows/release.yaml:56]: we should merged this and handle ^ in a follow up PR

**Review by arkodg — APPROVED** [review]
🚀

**Pull request merged**